### PR TITLE
Record usage of TupleElementNamesAttribute in VB

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
dotnet/roslyn-compiler

File: src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb [hunk 1]
```diff
@@ -1902,16 +1902,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
         Friend ReadOnly Property HasTupleNamesAttributes As Boolean
             Get
-                Return GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_TupleElementNamesAttribute__ctorTransformNames) IsNot Nothing
+                Dim constructorSymbol = TryCast(GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_TupleElementNamesAttribute__ctorTransformNames), MethodSymbol)
+                Return constructorSymbol IsNot Nothing AndAlso
+                       Binder.GetUseSiteInfoForWellKnownTypeMember(constructorSymbol, WellKnownMember.System_Runtime_CompilerServices_TupleElementNamesAttribute__ctorTransformNames,
+                                                                   embedVBRuntimeUsed:=False).DiagnosticInfo Is Nothing
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb]: >DiagnosticInfo [](start = 94, length = 14)

So, if the attribute ctor is obsolete with a warning this returns false? If that is the intended behavior, why?

Comment (AlekseyTs) [reply src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb]: > So, if the attribute ctor is obsolete with a warning this returns false? If that is the intended behavior, why?

There is no change around this scenario. Obsolete checks are performed separately and presence of an Obsolete attributes on a symbol does not have any effect on DiagnosticInfo. Also, in VB the DiagnosticInfo property never returns warnings (unlike in C#).

---
In reply to: [386555701](https://github.com/dotnet/roslyn/pull/42048#discussion_r386555701) [](ancestors = 386555701)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off for a small change.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off for a small change.

**Pull request merged**